includes/amp-helper-functions.php

```diff
@@ -579,23 +579,56 @@ function amp_get_schemaorg_metadata() {
 		),
 	);
 
-	/**
-	 * Filters the site icon used in AMP responses.
-	 *
-	 * In general the `get_site_icon_url` filter should be used instead.
-	 *
-	 * @since 0.3
-	 * @todo Why is the size set to 32px?
-	 *
-	 * @param string $site_icon_url
-	 */
-	$site_icon_url = apply_filters( 'amp_site_icon_url', get_site_icon_url( AMP_Post_Template::SITE_ICON_SIZE ) );
-	if ( $site_icon_url ) {
-		$metadata['publisher']['logo'] = array(
-			'@type'  => 'ImageObject',
-			'url'    => $site_icon_url,
-			'height' => AMP_Post_Template::SITE_ICON_SIZE,
-			'width'  => AMP_Post_Template::SITE_ICON_SIZE,
+	$custom_logo_id  = get_theme_mod( 'custom_logo' );
+	$max_logo_width  = 600;
+	$max_logo_height = 60;
+	if ( $custom_logo_id && 'attachment' === get_post_type( $custom_logo_id ) ) {
+		$custom_logo_img = wp_get_attachment_image_src( $custom_logo_id, 'full', false );
+		$is_proper_size  = (
+			isset( $custom_logo_img[0], $custom_logo_img[1], $custom_logo_img[2] )
+			&&
+			$custom_logo_img[1] <= $max_logo_width
+			&&
+			$custom_logo_img[2] <= $max_logo_height
+		);
+
+		if ( $is_proper_size ) {
```

Comment:
Good point. The new commit [04a504](https://github.com/Automattic/amp-wp/pull/1144/commits/04a504cd7f77bb4817283d5b81c9d97243b72f6d) simply passes the max width and height to `wp_get_attachment_image_src()`. That will get a smaller image size if possible. So there's now no need for `$is_proper_size`.

This has worked so far in testing it with the [Google Structured Data Testing Tool](https://search.google.com/structured-data/testing-tool/u/0/) (though I had to substitute [public images URLs](https://picsum.photos/250/250) for local image URLs, as the tool raises errors when it can't get the URL.

<img width="1428" alt="passes-max-width-height" src="https://user-images.githubusercontent.com/4063887/40259424-a6258da6-5abb-11e8-8b9c-ca6d6a991d74.png">